src/xo-app/settings/groups/index.js

```diff
@@ -0,0 +1,145 @@
+import ActionButton from 'action-button'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import { error } from 'notification'
+import { SelectSubject } from 'select-objects'
+import includes from 'lodash/includes'
+import keyBy from 'lodash/keyBy'
+import orderBy from 'lodash/orderBy'
+
+import {
+  addUserToGroup,
+  createGroup,
+  deleteGroup,
+  removeUserFromGroup,
+  subscribeGroups,
+  subscribeUsers
+} from 'xo'
+
+const refreshSubs = () => {
+  subscribeUsers.forceRefresh()
+  subscribeGroups.forceRefresh()
+}
+
+class UserDisplay extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      groups: {},
+      users: {}
+    }
+  }
+
+  componentWillMount () {
+    const unsubscribeUsers = subscribeUsers(users => {
+      users = keyBy(users, 'id')
+      this.setState({
+        users
+      })
+    })
+
+    this.componentWillUnmount = () => {
+      unsubscribeUsers()
+    }
+  }
+
+  _removeUser = () => {
+    const {id, group} = this.props
+    return removeUserFromGroup({user: id, group}).then(() => subscribeGroups.forceRefresh()).catch(err => error('Remove User', err.message || String(err)))
+  }
+
+  render () {
+    const { id } = this.props
+    const {
+      users
+    } = this.state
+    return <span>
+      {id && (users[id] && users[id].email) || <em>&lt;unknown user or group&gt;</em>}
+      {' '}
+      <ActionButton btnStyle='primary' size='small' icon='remove' handler={this._removeUser} />
+    </span>
+  }
+}
+
+export default class Groups extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      groups: []
+    }
+  }
+
+  componentWillMount () {
+    const unsubscribeGroups = subscribeGroups(groups => this.setState({groups: orderBy(groups, ['name'])}))
+
+    this.componentWillUnmount = () => {
+      unsubscribeGroups()
+    }
+  }
+
+  _createGroup = () => {
+    const { name } = this.refs
+    if (name) {
+      return createGroup({name: name.value}).then(() => subscribeGroups.forceRefresh()).catch(err => error('Create Group', err.message || String(err)))
+    }
+  }
+
+  _deleteGroup = id => deleteGroup({id}).then(() => subscribeGroups.forceRefresh()).catch(err => error('Delete Group', err.message || String(err)))
+
+  _getPredicate = users => {
+    return entity => entity.email && !includes(users, entity.id) // Entity is a user and is not already in list
+  }
+
+  _addUser = (user, group) => {
+    return addUserToGroup({user: user.id, group: group.id}).then(refreshSubs).catch(err => error('Add User', err.message || String(err)))
```

Comment:
I refresh both group sub and user sub after adding a user, but still, UserDisplay shows the added user as an "unknown user" for a few seconds after addition... I did not figure out why so far.